src/wgapi_userspace.rs

```diff
@@ -149,15 +149,23 @@ impl WireguardInterfaceApi for WireguardApiUserspace {
     /// Assign IP address to network interface.
     fn assign_address(&self, address: &IpAddrMask) -> Result<(), WireguardInterfaceError> {
         debug!("Assigning address {address} to interface {}", self.ifname);
+        let address_string = address.ip.to_string();
+
         let output = if cfg!(target_os = "macos") {
-            // On macOS, interface is point-to-point and requires a pair of addresses
-            let address_string = address.ip.to_string();
-            Command::new("ifconfig")
-                .args([&self.ifname, &address_string, &address_string])
-                .output()?
+            match address.ip {
+                IpAddr::V4(_) => {
+                    Command::new("ifconfig")
+                        // On macOS ipv4, interface is point-to-point and requires a pair of addresses
+                        .args([&self.ifname, "inet", &address_string, &address_string])
+                        .output()?
+                }
+                IpAddr::V6(_) => Command::new("ifconfig")
+                    .args([&self.ifname, "inet6", &address_string])
+                    .output()?,
+            }
         } else {
             Command::new("ifconfig")
-                .args([&self.ifname, &address.to_string()])
+                .args([&self.ifname, &address_string])
```

Comment:
I just noticed that this was using `&address.to_string()` and not `&address.ip.to_string()`  Let me know if this is not correct.  But fixing in case it was wrong